File: src/easy/min_deletion_size.zig
const std = @import("std");

pub fn minDeletionSize(strs: []const []const u8) i32 {
    var count: i32 = 0;
    const rows = strs.len;
    const cols = strs[0].len;
    var col: usize = 0;
    while (col < cols) : (col += 1) {
        var row: usize = 1;
        while (row < rows) : (row += 1) {
            if (strs[row - 1][col] > strs[row][col]) {
                count += 1;
                break;
            }
        }
    }
    return count;
}

File: src/easy/min_deletion_size.rs
pub fn min_deletion_size(strs: Vec<String>) -> i32 {
    let mut count: i32 = 0;
    let rows = strs.len();
    let cols = strs[0].len();
    let bytes: Vec<&[u8]> = strs.iter().map(|s| s.as_bytes()).collect();

    for col in 0..cols {
        let mut row = 1;
        while row < rows {
            if bytes[row - 1][col] > bytes[row][col] {
                count += 1;
                break;
            }
            row += 1;
        }
    }
    count
}
